### PR TITLE
feat(alerts): #MA-952 Reset all alerts from year transition or endpoint

### PR DESCRIPTION
**Pull request by qpic-cgi**
## Describe your changes
Update alerts delete endpoint to reset them by structure or/and start and end date or/and ids.
Use the same service on year transition to reset alets when it happens.

## Checklist tests
From Postman, try delete endpoints with the different parameters:
"/presences/alerts"

## Issue ticket number and link
[Jira - MA-952](https://entsupport.gdapublic.fr/browse/MA-952)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: presences/src/main/java/fr/openent/presences/service/AlertService.java [hunk 1]
```diff
@@ -46,10 +58,11 @@ public interface AlertService {
 
     /**
      * Delete alerts of given type for a student
-     * @param structureId   structure identifier
-     * @param studentId     student identifier
-     * @param type          alert type
-     * @return  {@link Future} of {@link JsonObject}
+     *
```
Comment (qpic-cgi) [inline presences/src/main/java/fr/openent/presences/service/AlertService.java:61]: indent

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java [hunk 2]
```diff
@@ -170,16 +216,16 @@ public Future<JsonObject> resetStudentAlertsCount(String structureId, String stu
                 " WHERE student_id = ? AND structure_id = ? AND type = ? ";
 
         JsonArray params = new JsonArray().add(studentId)
-                        .add(structureId)
```
Comment (qpic-cgi) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java:173]: indent

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java [hunk 3]
```diff
@@ -117,7 +163,7 @@ public void getStudentAlertNumberWithThreshold(String structureId, String studen
         Future<JsonObject> futureCount = Future.future();
 
         String queryThreshold = "SELECT alert_forgotten_notebook_threshold as threshold" +
-                " FROM "  + Presences.dbSchema + ".settings " +
```
Comment (qpic-cgi) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java:120]: indent

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java [hunk 4]
```diff
@@ -154,7 +200,7 @@ public void getStudentAlertNumberWithThreshold(String structureId, String studen
             if (event.succeeded()) {
                 JsonObject result = new JsonObject();
                 result.put("threshold", futureThreshold.result().getValue("threshold"));
-                result.put("count" ,futureCount.result().getValue("count"));
```
Comment (qpic-cgi) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java:157]: indent

**Review by alicedraillard — APPROVED** [review]

File: presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    public void start() {
+        super.start();
+        eb.consumer(this.getClass().getName(), this);
+    }
+
+    @Override
+    public void handle(Message<JsonObject> eventMessage) {
+        log.info(String.format("[<worker>Presences@%s] receiving", this.getClass().getSimpleName()));
+        log.info(String.format("%s%s Begin to process %s", ANSI_CYAN, RESET_ALERTS_FLAG, ANSI_RESET));
+        eventMessage.reply(new JsonObject().put("status", "ok"));
```
Comment (BastienBenard) [inline presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java]: magic string

**Review by BastienBenard — APPROVED** [review]

File: presences/src/main/java/fr/openent/presences/Presences.java [hunk 6]
```diff
@@ -136,4 +145,8 @@ public void start() throws Exception {
         vertx.deployVerticle(EventExportWorker.class, new DeploymentOptions().setConfig(config).setWorker(true));
     }
 
+    public static void launchResetAlertsWorker(EventBus eb, JsonObject params) {
+        eb.send(ResetAlertsWorker.class.getName(), params, new DeliveryOptions().setSendTimeout(1000 * 1000L));
```
Comment (JinDouang) [inline presences/src/main/java/fr/openent/presences/Presences.java:150]: Eb.request non ?

File: presences/src/main/java/fr/openent/presences/Presences.java [hunk 7]
```diff
@@ -136,4 +145,8 @@ public void start() throws Exception {
         vertx.deployVerticle(EventExportWorker.class, new DeploymentOptions().setConfig(config).setWorker(true));
     }
```
Comment (JinDouang) [inline presences/src/main/java/fr/openent/presences/Presences.java:148]: On a pas besoin de déployer un verticle ?

File: presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    public static final String ANSI_RESET = "\u001B[0m";
+    public static final String ANSI_CYAN = "\u001B[36m";
+    private static final String RESET_ALERTS_FLAG = "[PRESENCES:RESET_ALERTS]";
+
+
+    private final AlertService alertService = new DefaultAlertService();
+
+    @Override
+    public void start() {
+        super.start();
```
Comment (JinDouang) [inline presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java:24]: Parce que si tu ne déploies pas (cf commentaire sur la question), tu ne deploies pas le vertx de cette classe la donc comment le eb peut être exposable ?

File: presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        eb.consumer(this.getClass().getName(), this);
+    }
+
+    @Override
+    public void handle(Message<JsonObject> eventMessage) {
+        log.info(String.format("[<worker>Presences@%s] receiving", this.getClass().getSimpleName()));
+        log.info(String.format("%s%s Begin to process %s", ANSI_CYAN, RESET_ALERTS_FLAG, ANSI_RESET));
+        eventMessage.reply(new JsonObject().put(Field.STATUS, Field.OK));
```
Comment (JinDouang) [inline presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java:32]: Magic string

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: presences/src/main/java/fr/openent/presences/Presences.java (hunk 6 above)
Comment (qpic-cgi) [reply presences/src/main/java/fr/openent/presences/Presences.java:150]: je me suis posé la question, mais mon intellij ne me le met pas en déprécié

Comment (qpic-cgi) [reply presences/src/main/java/fr/openent/presences/Presences.java:150]: Effectivement, send sans retour n'est pas déprécié et fonctionne !

File: presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java (hunk 9 above)
Comment (qpic-cgi) [reply presences/src/main/java/fr/openent/presences/worker/ResetAlertsWorker.java:32]: où ?

File: presences/src/main/java/fr/openent/presences/Presences.java (hunk 7 above)
Comment (qpic-cgi) [reply presences/src/main/java/fr/openent/presences/Presences.java:147]: quel verticle ?

File: presences/src/main/java/fr/openent/presences/Presences.java (hunk 6 above)
Comment (JinDouang) [reply presences/src/main/java/fr/openent/presences/Presences.java:150]: https://vertx.io/docs/3.9.5/apidocs/io/vertx/core/eventbus/EventBus.html#send-java.lang.String-java.lang.Object-

Deprecated. use [request(String, Object, Handler)](https://vertx.io/docs/3.9.5/apidocs/io/vertx/core/eventbus/EventBus.html#request-java.lang.String-java.lang.Object-io.vertx.core.Handler-) instead

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: presences/src/main/java/fr/openent/presences/Presences.java (hunk 6 above)
Comment (qpic-cgi) [reply presences/src/main/java/fr/openent/presences/Presences.java:150]: https://vertx.io/docs/3.9.5/apidocs/io/vertx/core/eventbus/EventBus.html#send-java.lang.String-java.lang.Object-:~:text=be%20used%20fluently-,send,-EventBus%C2%A0send(

Comment (JinDouang) [reply presences/src/main/java/fr/openent/presences/Presences.java:150]: ohoh bien vu! Je resolve

**Review by JinDouang — APPROVED** [review]

**Pull request merged**